dvc/utils/__init__.py

```diff
@@ -249,3 +252,83 @@ def walk_files(directory):
     for root, _, files in os.walk(str(directory)):
         for f in files:
             yield os.path.join(root, f)
+
+
+def colorize(message, color=None):
+    """Returns a message in a specified color."""
+    if not color:
+        return message
+
+    colors = {
+        "green": colorama.Fore.GREEN,
+        "yellow": colorama.Fore.YELLOW,
+        "blue": colorama.Fore.BLUE,
+        "red": colorama.Fore.RED,
+    }
+
+    return "{color}{message}{nc}".format(
+        color=colors.get(color, ""), message=message, nc=colorama.Fore.RESET
+    )
+
+
+def boxify(message, border_color=None):
+    """Put a message inside a box.
+
+    Args:
+        message (unicode): message to decorate.
+        border_color (unicode): name of the color to outline the box with.
+    """
+    lines = message.split("\n")
+    max_width = max(_visual_width(line) for line in lines)
+
+    padding_horizontal = 5
+    padding_vertical = 1
+
+    box_size_horizontal = max_width + (padding_horizontal * 2)
+
+    chars = {"corner": "+", "horizontal": "-", "vertical": "|", "empty": " "}
+
+    margin = "{corner}{line}{corner}\n".format(
+        corner=chars["corner"], line=chars["horizontal"] * box_size_horizontal
+    )
+
+    padding_lines = [
+        "{border}{space}{border}\n".format(
+            border=colorize(chars["vertical"], color=border_color),
```

Comment:
Should we still colorize it if output is not a tty though?

Reply:
I'd say yes, when logging something colorized, the formatter would take care of stripping the output of it's not tty.